web/src/reducers/dirty.js

```diff
@@ -52,9 +52,26 @@ const up = updates =>
 const dirty = (state = initialState, action) => {
   switch (action.type) {
     case SAVE_APD_SUCCESS:
-    case SELECT_APD:
       return initialState;
 
+    case SELECT_APD:
+      // When we select an APD, stash off all the activity keys so we'll
+      // have them to iterate over later, if the APD years change (see
+      // below for more explanation of why).
+      return u(
+        {
+          data: {
+            activities: {
+              byKey: action.apd.activities.reduce(
+                (byKey, activity) => ({ ...byKey, [activity.key]: false }),
```

Comment:
What does the `false` indicate?

EDIT: Never mind, I figured it out (more or less) by reading the tests, which means they're good tests, Brent.